flake.nix

```diff
@@ -319,9 +330,9 @@
               cfg = config.homestakeros;
 
               # Function to get the active client
-              getActiveClient = clients:
+              getActiveClient = clients: path:
                 let
-                  enabledClients = builtins.filter (name: cfg.${name}.enable) clients;
+                  enabledClients = builtins.filter (name: path.${name}.enable) clients;
                   numEnabledClients = builtins.length enabledClients;
                 in
                 if numEnabledClients == 1 then
```

Comment:
`getActiveClient` should be `getActiveClients` and always an array. An empty array is not a throwable error: it just means that no client is enabled. Usually, this also avoids the need to do a null check, as in the other comment: if the list is empty, then the iteration has length 0, so nothing gets iterated, effectively just skipping the service enablation without the need of a type check.